### PR TITLE
Regridder interface

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
Move functions from `coupler_utils/regridder.jl` to `Regridder` module in `src/`, and ensure that they are thoroughly commented and tested.
Additionally, create a `Utilities` module in `src/` containing functions commonly used by other coupler modules.

Part of SDI #134

## To-do
PHASE 1 - moving and documentation
- [x] Copy file(s) to module in `src/`
- [x] Add module to `ClimaCoupler.jl`
- [x] Docstring for each function
  - [x] Add input/output types in docstring if not type annotated
  - [x] Add docstring info to `docs/src`, `docs/make.jl`
- [x] Add exports for externally-used functions
- [x] Flag potentially-removable parts of code

PHASE 2 - testing
- [x] Create test file in `test/`
- [x] Unit test for each exported function in module
  - [x] Test simplest functions first
- [x] Integration tests as needed
- [x] Create test file in `test/mpi_tests` and add any tests that use MPI

PHASE 3 - cleanup
- [x] Remove flagged parts of code
  - [x] Test after each one
- [x] Check all imports are necessary
- [x] Formatting
- [x] Read through comments

### CODE REVIEW

**Comment by juliasloan25** [comment]
bors try

File: experiments/AMIP/moist_mpi_earth/coupler_driver.jl [hunk 1]
```diff
@@ -64,6 +64,16 @@ using ClimaCore: InputOutput
...
+# modify parsed args for fast testing from REPL #hide
+# parsed_args["coupled"] =  true #hide
+# parsed_args["moist"]= "equil" #hide
+# parsed_args["vert_diff"] =  true #hide
+# parsed_args["rad"] =  "gray" #hide
+# parsed_args["microphy"] =  "0M" #hide
+# parsed_args["energy_check"] = true
+# parsed_args["precip_model"] =  "0M" #hide
+# parsed_args["mode_name"] =  "slabplanet" #hide
+
```
Comment (LenkaNovak) [inline experiments/AMIP/moist_mpi_earth/coupler_driver.jl:76]: I've just found `isinteractive()`, which we can use here as a conditional for running from the REPL, so we can leave this uncommented.

File: test/TestHelper.jl [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Create an NCDataset with lat/lon dimensions containing the value `val` for the
+variable `varname`, and store it at `path`.
+
+# Arguments
+- `FT`: [DataType] Float type.
+- `path`: [String] location to store output datafile.
+- `varname`: [Symbol] variable name.
+- `val`: [FT] value to store as `varname` at all indices.
+"""
+function gen_ncdata!(FT, path, varname, val)
```
Comment (LenkaNovak) [inline test/TestHelper.jl:58]: The bang is not consistent with the docstring. AFAIK the convention is only to use bangs when arguments are mutated, so I would remove it here.

**Review by LenkaNovak — APPROVED** [review]
Hi @juliasloan25 , looks great! I've only left a couple of small comments, after resolving which feel free to merge. Thanks!

**Pull request merged**